task/control-plane-agent/src/mgs_common.rs

```diff
@@ -405,6 +406,72 @@ impl MgsCommon {
             Err(e) => Err(e.into()),
         }
     }
+
+    #[cfg(not(feature = "vpd"))]
+    pub(crate) fn vpd_lock_status_all(
+        &self,
+        _buf: &mut [u8],
+    ) -> Result<usize, SpError> {
+        Err(SpError::Vpd(GatewayVpdError::NotImplemented))
+    }
+
+    #[cfg(feature = "vpd")]
+    pub(crate) fn vpd_lock_status_all(
+        &self,
+        buf: &mut [u8],
+    ) -> Result<usize, SpError> {
+        use task_vpd_api::{Vpd, VpdError};
+        task_slot!(VPD, vpd);
+
+        let vpd = Vpd::from(VPD.get_task_id());
+        let cnt = vpd.num_vpd_devices();
+
+        for (i, entry) in buf.iter_mut().enumerate().take(cnt) {
+            // The API takes a `u8` and we're bounded by the number of devices
+            // in the array which is also going to be less than `u8`
+            let idx = match u8::try_from(i) {
+                Ok(v) => v,
+                Err(_) => unreachable!(),
+            };
+            *entry = match vpd.is_locked(idx) {
+                Ok(v) => v.into(),
+                Err(e) => {
+                    return Err(SpError::Vpd(match e {
+                        VpdError::InvalidDevice => {
```

Comment:
Should we implement `From<VpdError> for GatewayVpdError`?

(It would have to go in `vpd-api`, so maybe that's too much plumbing)